www/js/diary/infinite_scroll_list.js

```diff
@@ -232,9 +235,9 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
   $scope.$on("enketo.noteAddition", (e, addition, scrollElement) => {
     $scope.$apply(() => {
       // TODO support places
-      const matchingTimelineEntry = $scope.data.displayTrips.find((trip) => {
-        return addition.data.start_ts >= ~~trip.start_ts && addition.data.start_ts <= trip.end_ts
-      });
+      const matchingTimelineEntry = $scope.data.displayTrips.find((trip) => 
+        InputMatcher.validUserInputForTrip(trip, addition, false)
+      );
```

Comment:
Yes. This is what I meant. No need to copy and paste code and come up with duplicate and slightly different implementations.